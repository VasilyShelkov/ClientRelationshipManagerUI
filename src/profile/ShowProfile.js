import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';

import Paper from 'material-ui/Paper';
import { List, ListItem } from 'material-ui/List';
import Avatar from 'material-ui/Avatar';
import Divider from 'material-ui/Divider';
import RaisedButton from 'material-ui/RaisedButton';
import Chip from 'material-ui/Chip';

import EditIcon from 'material-ui/svg-icons/content/create';
import ProfileIcon from 'material-ui/svg-icons/social/person';
import PhoneIcon from 'material-ui/svg-icons/communication/phone';
import EmailIcon from 'material-ui/svg-icons/communication/email';
import SecurityIcon from 'material-ui/svg-icons/hardware/security';
import UpdatedIcon from 'material-ui/svg-icons/action/update';
import { cyan500 } from 'material-ui/styles/colors';

import EditPassword from './EditPassword';
import {
  editProfile, editProfilePassword, cancelEditProfilePassword, editProfilePasswordSuccess
} from './profileActions';

export const ShowProfile = ({
  userId, firstName, lastName, email, phone, updatedAt, editingPassword,
  onEditProfile, onEditProfilePassword, onEditProfilePasswordSuccess,
  onCancelEditProfilePassword
}) => (
  <Paper zDepth={2} style={{ paddingBottom: '10px' }}>
    <div style={{ textAlign: 'center' }}>
      <ProfileIcon style={{ height: '100px', width: '100px' }} color={cyan500} /> 
      <h2>{firstName} {lastName}</h2>
    </div>


    <List >
      <Divider />
    
      <ListItem
        leftAvatar={<Avatar icon={<EmailIcon />} backgroundColor={cyan500} />}
        primaryText={email}
        disabled
      />
      <ListItem
        leftAvatar={<Avatar icon={<PhoneIcon />} backgroundColor={cyan500} />}
        primaryText={phone}
        disabled
      />

      <Divider />

      {
        editingPassword ?
          <EditPassword 
            userId={userId}
            handleProfilePasswordSuccess={onEditProfilePasswordSuccess}
            handleCancelEditProfilePassword={onCancelEditProfilePassword}
          />
        :
          <ListItem
            leftAvatar={<Avatar icon={<SecurityIcon />} backgroundColor={cyan500} />}
            rightIcon={<EditIcon />}
            primaryText="Password"
            secondaryText="**********"
            onClick={onEditProfilePassword}
          />
      }

      <Divider />
    </List>


    <div className="Profile__last-updated row justify-content-center">
      <Chip>
        <Avatar size={32} icon={<UpdatedIcon />} backgroundColor={cyan500} />
        Last Updated: {moment(updatedAt).fromNow()}
      </Chip>
    </div>
    <div className="Profile__cta row justify-content-center">
      <RaisedButton
        primary
        label="Edit Profile"
        icon={<EditIcon />}
        onClick={onEditProfile}
      />
    </div>
  </Paper>
);

const mapStateToProps = state => ({
  editingPassword: state.profile.editingPassword
});

const mapDispatchToProps = dispatch => ({
  onEditProfile: () => dispatch(editProfile()),
  onEditProfilePassword: () => dispatch(editProfilePassword()),
  onEditProfilePasswordSuccess: () => dispatch(editProfilePasswordSuccess()),
  onCancelEditProfilePassword: () => dispatch(cancelEditProfilePassword()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ShowProfile);
