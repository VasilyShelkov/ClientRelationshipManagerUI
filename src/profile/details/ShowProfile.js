import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';

import Paper from 'material-ui/Paper';
import { List, ListItem } from 'material-ui/List';
import Divider from 'material-ui/Divider';
import RaisedButton from 'material-ui/RaisedButton';
import Chip from 'material-ui/Chip';
import Avatar from 'material-ui/Avatar';

import EditIcon from 'material-ui/svg-icons/content/create';
import ProfileIcon from 'material-ui/svg-icons/social/person';
import PhoneIcon from 'material-ui/svg-icons/communication/phone';
import EmailIcon from 'material-ui/svg-icons/communication/email';
import SecurityIcon from 'material-ui/svg-icons/hardware/security';
import UpdatedIcon from 'material-ui/svg-icons/action/update';
import SuccessfulUpdateIcon from 'material-ui/svg-icons/action/done';
import SuccessfullyCreatedIcon from 'material-ui/svg-icons/action/check-circle';
import {
  cyan500,
  green900,
  green700,
  lightGreen300,
} from 'material-ui/styles/colors';
import { ProtectedIcon } from '../../app/icons';
import toMaterialStyle from 'material-color-hash';

import Notification from '../../shared/Notification';
import EditPassword from './EditPassword';
import {
  editProfile,
  editProfilePassword,
  cancelEditProfilePassword,
  removeProfileNotification,
} from '../profileActions';
import { EDIT_IN_PROGRESS } from '../profileReducer';

export const ShowProfile = ({
  id,
  firstName,
  lastName,
  email,
  phone,
  updated_at,
  protectedNamesLimit,
  newUserNotification,
  editingPassword,
  editSuccessProfileNotification,
  onEditProfile,
  onEditProfilePassword,
  onCancelEditProfilePassword,
  onRemoveNotification,
}) => (
  <Paper zDepth={2} style={{ paddingBottom: '10px', margin: '20px 0px' }}>
    <div style={{ textAlign: 'center', padding: '0px 10px' }}>
      <div style={{ height: '85px' }}>
        <ProfileIcon
          style={{
            height: '100px',
            width: '100px',
            position: 'relative',
            left: '20px',
          }}
          color={cyan500}
        />
        <Avatar
          style={{
            ...toMaterialStyle(
              `${firstName[0].toUpperCase()}${lastName[0].toUpperCase()}`,
            ),
            position: 'relative',
            top: '-60px',
            right: '50px',
          }}
        >
          {firstName[0].toUpperCase()}
          {lastName[0].toUpperCase()}
        </Avatar>
      </div>
      <h2>
        {firstName} {lastName}
      </h2>
      <Notification
        message={newUserNotification}
        zDepth={3}
        backgroundColor={green700}
        icon={
          <SuccessfullyCreatedIcon
            className="Form__notification__icon"
            style={{ color: 'white' }}
          />
        }
      />
    </div>

    <List>
      <Divider />

      <ListItem
        leftAvatar={
          <Avatar icon={<ProtectedIcon />} backgroundColor={cyan500} />
        }
        primaryText={`Limit: ${protectedNamesLimit}`}
        disabled
      />
      <ListItem
        leftAvatar={<Avatar icon={<EmailIcon />} backgroundColor={cyan500} />}
        primaryText={email}
        disabled
      />
      <ListItem
        leftAvatar={<Avatar icon={<PhoneIcon />} backgroundColor={cyan500} />}
        primaryText={phone}
        disabled
      />

      <Divider />

      {editingPassword ? (
        <EditPassword
          userId={id}
          editInProgress={editingPassword === EDIT_IN_PROGRESS}
          handleCancelEditProfilePassword={onCancelEditProfilePassword}
        />
      ) : (
        <ListItem
          id="resetPassword"
          leftAvatar={
            <Avatar icon={<SecurityIcon />} backgroundColor={cyan500} />
          }
          rightIcon={<EditIcon />}
          primaryText="Password"
          secondaryText="**********"
          onClick={onEditProfilePassword}
        />
      )}

      <Divider />
    </List>

    <div className="Profile__meta-info">
      <div className="row justify-content-center">
        {editSuccessProfileNotification && (
          <Chip
            id="editProfileSuccess"
            style={{ marginBottom: '10px' }}
            backgroundColor={lightGreen300}
            onRequestDelete={onRemoveNotification}
            onTouchTap={onRemoveNotification}
          >
            <Avatar
              size={32}
              icon={<SuccessfulUpdateIcon />}
              color={lightGreen300}
              backgroundColor={green900}
            />
            {editSuccessProfileNotification}
          </Chip>
        )}
      </div>
      <div className="row justify-content-center">
        <Chip>
          <Avatar size={32} icon={<UpdatedIcon />} backgroundColor={cyan500} />
          <strong>Last Updated</strong>: {moment(updated_at).fromNow()}
        </Chip>
      </div>
    </div>
    <div className="Profile__cta row justify-content-center">
      <RaisedButton
        primary
        label="Edit Profile"
        icon={<EditIcon />}
        onClick={onEditProfile}
      />
    </div>
  </Paper>
);

const mapStateToProps = state => ({
  editSuccessProfileNotification: state.profile.notification.profile,
  editingPassword: state.profile.editing.password,
  newUserNotification: state.profile.notification.newUser,
});

const mapDispatchToProps = dispatch => ({
  onEditProfile: () => dispatch(editProfile()),
  onEditProfilePassword: () => dispatch(editProfilePassword()),
  onCancelEditProfilePassword: () => dispatch(cancelEditProfilePassword()),
  onRemoveNotification: () => dispatch(removeProfileNotification()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ShowProfile);
