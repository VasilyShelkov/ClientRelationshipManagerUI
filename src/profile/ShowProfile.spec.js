import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import RaisedButton from 'material-ui/RaisedButton';
import { shallow } from 'enzyme';
import ShowProfile from './ShowProfile';

describe('src/profile/Profile.js', () => {
  it('renders the Home page', () => {
    expect(renderer.create(
      <MuiThemeProvider>
        <ShowProfile
          firstName="Vasily"
          lastName="Shelkov"
          email="vasilydshelkov@gmail.com"
          phone="07123456789"
        />
      </MuiThemeProvider>
    ).toJSON())
      .toMatchSnapshot();
  });

  it('calls the onEditProfile when the button is clicked', () => {
    const onEditProfileMock = jest.fn();
    const wrapper = shallow(
      <ShowProfile
        firstName="Vasily"
        lastName="Shelkov"
        email="vasilydshelkov@gmail.com"
        phone="07123456789"
        onEditProfile={onEditProfileMock}
      />
    );

    const renderedEditButtonClick = wrapper.find(RaisedButton).prop('onClick');
    renderedEditButtonClick();

    expect(onEditProfileMock.mock.calls.length).toBe(1);
  });
});
