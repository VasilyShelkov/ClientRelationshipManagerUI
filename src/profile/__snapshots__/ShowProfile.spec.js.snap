exports[`src/profile/Profile.js renders the Home page 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
             0 3px 10px rgba(0, 0, 0, 0.23)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }>
  <div
    className="Profile__details">
    <div
      className="row">
      <div
        className="col-xs-12 col-sm-6">
        <h4
          className="Profile__details__field">
          First name:
        </h4>
        Vasily
      </div>
      <div
        className="col-xs-12 col-sm-6">
        <h4
          className="Profile__details__field">
          Last name:
        </h4>
        Shelkov
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-xs-12 col-sm-6">
        <h4
          className="Profile__details__field">
          Email:
        </h4>
        vasilydshelkov@gmail.com
      </div>
      <div
        className="col-xs-12 col-sm-6">
        <h4
          className="Profile__details__field">
          phone:
        </h4>
        07123456789
      </div>
    </div>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": "2px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "minWidth": "100%",
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }>
    <button
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "#00bcd4",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "100%",
        }
      }
      tabIndex={0}
      type="button">
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": 36,
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }>
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fill": "#ffffff",
                "height": 24,
                "marginLeft": 12,
                "marginRight": 0,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "middle",
                "width": 24,
              }
            }
            viewBox="0 0 24 24">
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z" />
          </svg>
          <span
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 8,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }>
            Edit profile
          </span>
        </div>
      </div>
    </button>
  </div>
</div>
`;
